pkg/infra/pulumi_aws/deploylib.ts

```diff
@@ -179,62 +179,62 @@ export class CloudCCLib {
-                    fromPort: 0,
-                    protocol: '-1',
-                    self: true,
-                    toPort: 0,
-                },
-                {
-                    description: 'For EKS control plane',
-                    cidrBlocks: ['0.0.0.0/0'],
-                    fromPort: 9443,
-                    protocol: 'TCP',
-                    self: true,
-                    toPort: 9443,
-                },
-            ],
-        })
-        this.sgs = new Array(klothoSG.id)
-
-        pulumi.output(this.klothoVPC.privateSubnets).apply((ps) => {
-            const cidrBlocks: any = ps.map((subnet) => subnet.subnet.cidrBlock)
-            new aws.ec2.SecurityGroupRule(`${this.name}-private-ingress`, {
-                type: 'ingress',
-                cidrBlocks: cidrBlocks,
-                fromPort: 0,
-                protocol: '-1',
-                toPort: 0,
-                securityGroupId: klothoSG.id,
-            })
-        })
-
-        pulumi.output(this.klothoVPC.publicSubnets).apply((ps) => {
-            const cidrBlocks: any = ps.map((subnet) => subnet.subnet.cidrBlock)
-            new aws.ec2.SecurityGroupRule(`${this.name}-public-ingress`, {
-                type: 'ingress',
-                cidrBlocks: cidrBlocks,
-                fromPort: 0,
-                protocol: '-1',
-                toPort: 0,
-                securityGroupId: klothoSG.id,
+        pulumi
+            .all([this.klothoVPC.privateSubnets || [], this.klothoVPC.publicSubnets || []])
+            .apply(([privateSubnets, publicSubnets]) => {
+                const privateCidrBlocks: any = privateSubnets.map(
+                    (subnet) => subnet.subnet.cidrBlock
+                )
+                const publicCidrBlocks: any = publicSubnets.map((subnet) => subnet.subnet.cidrBlock)
+                const klothoSG = new aws.ec2.SecurityGroup(sgName, {
+                    name: sgName,
+                    vpcId: this.klothoVPC.id,
+                    egress: [
+                        {
+                            cidrBlocks: ['0.0.0.0/0'],
+                            description: 'Allows all outbound IPv4 traffic.',
+                            fromPort: 0,
+                            protocol: '-1',
+                            toPort: 0,
+                        },
+                    ],
+                    ingress: [
+                        {
+                            description:
+                                'Allows inbound traffic from network interfaces and instances that are assigned to the same security group.',
+                            fromPort: 0,
+                            protocol: '-1',
+                            self: true,
+                            toPort: 0,
+                        },
+                        {
+                            description: 'For EKS control plane',
```

Comment:
Should this only get added if there's an eks being set up?

Reply:
well we arent smart enough for that yet because we dont split up exec units by type. So ideally yes, but its not really doable without a major refactor